File: backend/src/main/java/vn/edu/fpt/doghandbook/backend/service/BreedService.java
package vn.edu.fpt.doghandbook.backend.service;

import vn.edu.fpt.doghandbook.backend.dto.response.BreedCompareResponse;
import vn.edu.fpt.doghandbook.backend.dto.response.BreedResponse;
import vn.edu.fpt.doghandbook.backend.dto.response.DevelopmentStageResponse;
import vn.edu.fpt.doghandbook.backend.dto.response.PageResponse;
import vn.edu.fpt.doghandbook.backend.dto.request.BreedRequest;

import java.util.List;

public interface BreedService {

    PageResponse<BreedResponse> getAll(int page, int size, String search);

    BreedResponse getById(Integer id);

    BreedResponse create(BreedRequest request, Integer createdByUserId);

    BreedResponse update(Integer id, BreedRequest request);

    void delete(Integer id);

    BreedCompareResponse compare(List<Integer> breedIds);

    List<DevelopmentStageResponse> getDevelopmentStages(Integer breedId);
}

File: web-admin/src/services/breedService.js
import api from './api';

export const breedService = {
  getAll: (page = 0, size = 10, search = '') =>
    api.get(`/breeds?page=${page}&size=${size}&search=${search}&sort=updatedAt,desc&sort=createdAt,desc`),
  getById: (id) => api.get(`/breeds/${id}`),
  create: (data) => api.post('/breeds', data),
  update: (id, data) => api.put(`/breeds/${id}`, data),
  delete: (id) => api.delete(`/breeds/${id}`),
  compare: (breedIds) => api.post('/breeds/compare', { breedIds }),
  getStages: (id) => api.get(`/breeds/${id}/development-stages`),
};
